staging/src/k8s.io/code-generator/cmd/client-gen/generators/generator_for_type.go

```diff
@@ -425,146 +405,162 @@ var interfaceTemplate4 = `
 }
 `
 
-// template for the struct that implements the type's interface
-var structType = `
-// $.type|privatePlural$ implements $.type|public$Interface
-type $.type|privatePlural$ struct {
-	*$.TypeClient|raw$[*$.resultType|raw$]
-}
-`
-
-var structTypeWithList = `
-// $.type|privatePlural$ implements $.type|public$Interface
-type $.type|privatePlural$ struct {
-	*$.TypeClientWithList|raw$[*$.resultType|raw$, *$.resultType|raw$List]
-}
-`
-
-var structTypeWithApply = `
-// $.type|privatePlural$ implements $.type|public$Interface
-type $.type|privatePlural$ struct {
-	*$.TypeClientWithApply|raw$[*$.resultType|raw$, *$.inputApplyConfig|raw$]
-}
-`
-
-var structTypeWithListAndApply = `
-// $.type|privatePlural$ implements $.type|public$Interface
-type $.type|privatePlural$ struct {
-	*$.TypeClientWithListAndApply|raw$[*$.resultType|raw$, *$.resultType|raw$List, *$.inputApplyConfig|raw$]
-}
-`
+// struct and constructor variants
+const (
+	namespaced    = 0
+	noListOrApply = 0
+	nonNamespaced = 1 << iota
+	withList
+	withApply
+)
 
-var newStructNamespaced = `
-// new$.type|publicPlural$ returns a $.type|publicPlural$
-func new$.type|publicPlural$(c *$.GroupGoName$$.Version$Client, namespace string) *$.type|privatePlural$ {
-	return &$.type|privatePlural${
-		generic.NewNamespaced[*$.resultType|raw$](
-			"$.type|resource$",
-			c.RESTClient(),
-			$.schemeParameterCodec|raw$,
-			namespace,
-			func() *$.resultType|raw$ { return &$.resultType|raw${} }),
+// struct declarations
+var structType = []string{
+	noListOrApply: `
```

Comment:
I’ll add some comments. I quite like explicit-index initialisation because it enforces index unicity, which helps catch errors. The slice in this instance isn’t sparse, the indices are contiguous (I’ll mention that in the comments).